src/pages/signin/SignInPage.js

```diff
@@ -44,16 +46,32 @@ const propTypes = {
         // Error to display when there is a session error returned
         authToken: PropTypes.string,
     }),
+
+    route: PropTypes.shape({
+        params: PropTypes.shape({
+            validateCode: PropTypes.string,
+        }),
+    }),
 };
 
 const defaultProps = {
     account: {},
     session: {},
     credentials: {},
+    route: {
+        params: {},
+    },
 };
 
 class SignInPage extends Component {
     componentDidMount() {
+        // If we've redirected to the signin page with a validation code, validate the login
+        if (_.isMatch(this.props.route, {name: 'ValidateLogin'})) {
+            const accountID = Number(lodashGet(this.props.route, 'params.accountID', 0));
+            const validateCode = lodashGet(this.props.route, 'params.validateCode', '');
+            validateLogin(accountID, validateCode);
```

Comment:
I'm confused about why `validateLogin` is being called a second time. I thought it was already called in `AuthScreens`, then it validates the login, and redirects to the settings page. Can you help me understand the code flow and what I'm missing?

Reply:
Ah, yes, so there's two routes for validation here depending on what state the user is in : 

**1. User is not signed in when clicking on validation link:** 
 - We navigate to PublicScreens.js where we see that we have a public route matching `ValidateLogin` and redirect to the signin page where we run `User.validateLogin`

**2. User is already logged in when clicking on the validation link**
- We go directly to AuthScreens.js, and run `User.validateLogin` in `componentDidMount()`, we redirect the user to Settings > Profile to show that their login has been updated